.github/workflows/npm-build-common-multiple-nodes-rabbitmq.yml

```diff
@@ -0,0 +1,107 @@
+name: reusable task to run builds
+
+on:
+  workflow_call:
+    # Inputs the workflow accepts.
+    inputs:
+      branch:
+        required: true
+        type: string
+      use-slack:
+        description: 'Should there be a slack notification'
+        type: boolean
+        default: true
+      package-name:
+        description: 'A package name to be used in the slack notification'
+        type: string
+        required: true
+      pull-request-url:
+        description: 'Link to the pull request'
+        required: true
+        type: string
+      build-command:
+        description: 'Should there be a build command'
+        type: string
+        default: 'build'
+
+    secrets:
+      git-token:
+        description: 'A git token passed from the caller workflow'
+        required: true
+      npm-token:
+        description: 'A npm token passed from the caller workflow'
+        required: true
+      slack-webhookurl:
+        description: 'A slack webhook url for status notification'
+        required: true
+
+jobs:
+  npm-build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [ 20.x, 22.x ]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          registry-url: 'https://registry.npmjs.org'
+          token: ${{ secrets.git-token }}
+      - name: Start RabbitMQ
```

Comment:
can you explain the file?

Reply:
it in use by https://github.com/TechSeeDev/jackrabbit/blob/master/.github/workflows/npm-build.yml

Reply:
i got that, but why you add a dedicated action for rabbit?